pullrequests.go

```diff
@@ -572,6 +579,32 @@ func (api *PullRequestsAPI) GetPullRequest(pullRequestNumber int) (*PullRequestR
 	return &pullRequest, nil
 }
 
+// MergePullRequest merges a pull request using the squash method.
+func (api *PullRequestsAPI) MergePullRequest(pullRequestNumber int) (*MergeRequestResponse, error) {
```

Comment:
There'll be times we want to do a merge commit, like merging from release to develop. You could create two methods or add a parameter.

If you use a parameter here's an example of using a "named type" based on string:

https://github.com/judwhite/ghapi/blob/cd279dd1b5fe4be127e125f0ce8d31f38484352b/status.go#L9-L21

I'm 👍 with either approach.